core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/ColumnPageWrapper.java

```diff
@@ -73,11 +113,16 @@ public int fillVector(int[] filteredRowId, ColumnVectorInfo[] vectorInfo, int ch
     if (null != localDictionary) {
       return localDictionary
           .getDictionaryValue(CarbonUtil.getSurrogateInternal(columnPage.getBytes(rowId), 0, 3));
-    } else if (columnType == ColumnType.COMPLEX_PRIMITIVE && this.isAdaptiveComplexPrimitive()) {
-      if (columnPage.getNullBits().get(rowId)) {
+    } else if (columnType == ColumnType.COMPLEX_PRIMITIVE
```

Comment:
In case of explict sorted data will be return based on invertedIndexReverse can u please check the same, Please refer UnsafeFixedLengthDimensionDataChunkStore.java getRowMethod